src/Features/Core/Portable/Wrapping/InitializerExpression/AbstractInitializerExpressionWrapper.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Indentation;
+
+namespace Microsoft.CodeAnalysis.Wrapping.InitializerExpression
+{
+    internal abstract partial class AbstractInitializerExpressionWrapper<
+        TListSyntax,
+        TListItemSyntax>
+        : AbstractSyntaxWrapper
+        where TListSyntax : SyntaxNode
+        where TListItemSyntax : SyntaxNode
+    {
+        protected abstract string Unwrap_list { get; }
+        protected abstract string Wrap_long_list { get; }
+        protected abstract string Unwrap_all_items { get; }
+        protected abstract string Wrap_every_item { get; }
+        protected abstract string Indent_all_items { get; }
+
+        protected abstract bool DoWrapInitializerOpenBrace { get; }
+
+        protected AbstractInitializerExpressionWrapper(IIndentationService indentationService)
+            : base(indentationService)
+        {
+        }
+
+        protected abstract TListSyntax TryGetApplicableList(SyntaxNode node);
+        protected abstract SeparatedSyntaxList<TListItemSyntax> GetListItems(TListSyntax listSyntax);
+
+        public override async Task<ICodeActionComputer> TryCreateComputerAsync(
+            Document document, int position, SyntaxNode declaration, CancellationToken cancellationToken)
+        {
+            var listSyntax = TryGetApplicableList(declaration);
+            if (listSyntax == null)
```

Comment:
If you're familiar with the new C# 8 nullable annotations, you may want to add them for this file. if not, that's fine too. Our pattern is to put `#nullable enable` under the copyright header on the top for new files, at least.